src/abc_atlas_access/abc_atlas_cache/cloud_cache.py

```diff
@@ -0,0 +1,222 @@
+
+        # the class users are actually interacting with
+        # (for warning message purposes)
+        if ui_class_name is None:
+            self._user_interface_class = type(self).__name__
+        else:
+            self._user_interface_class = ui_class_name
+
+        self._manifest = None
+        self._manifest_name = None
+
+        self._cache_dir = cache_dir
+
+        self._manifest_file_names = self._list_all_manifests()
+
+    # ====================== BasicLocalCache properties =======================
+
+    @property
+    def ui(self):
+        return self._user_interface_class
+
+    @property
+    def current_manifest(self) -> Union[None, str]:
+        """The name of the currently loaded manifest"""
+        return self._manifest_name
+
+    @property
+    def manifest_prefix(self) -> str:
+        """On-line prefix for manifest files"""
+        return '/releases/'
+
+    @property
+    def version(self) -> str:
+        """The version of the dataset currently loaded"""
+        return self._manifest.version
+
+    @property
+    def metadata_file_names(self) -> list:
+        """List of metadata file names associated with this dataset"""
+        return self._manifest.list_metadata
+
+    @property
+    def manifest_file_names(self) -> list:
+        """Sorted list of manifest file names associated with this dataset
+        """
+        return self._manifest_file_names
+
+    @property
+    def latest_manifest_file(self) -> str:
+        """parses on-line available manifest files for semver string
+        and returns the latest one
+        self.manifest_file_names are assumed to be of the form
+        '<version>/manifest.json'
+
+        Returns
+        -------
+        str
+            the filename whose semver string is the latest one
+        """
+        return self._find_latest_file(self.manifest_file_names)
+
+    @property
+    def cache_dir(self) -> str:
+        """Return the cache directory path.
+
+        Returns
+        -------
+        str
+            Full cache directory path
+        """
+        return self._cache_dir
+
+    # ====================== BasicLocalCache methods ==========================
+
+    @abstractmethod
+    def _list_all_manifests(self) -> list:
+        """
+        Return a list of all of the file names of the manifests associated
+        with this dataset
+        """
+        raise NotImplementedError()
+
+    def list_all_downloaded_manifests(self) -> list:
+        """
+        Return a list of all of the manifest files that have been
+        downloaded for this dataset
+        """
+        output = [x for x in os.listdir(self._cache_dir)
+                  if re.fullmatch(
+                      pattern="releases/*/manifest.json", string=x)]
+        output.sort()
+        return output
+
+    def _find_latest_file(self, file_name_list: List[str]) -> str:
+        """
+        Take a list of files named like
+
+        {blob}_v{version}.json
```

Comment:
Is this the correct specification? It looks like this method searches files named like
```
{blob}/{date}/manifest.json
```

Reply:
Haven't really started work on the cache object yet. Got pulled into this PR due to me being sloppy.